rct229/rulesets/ashrae9012019/ruleset_functions/get_energy_required_to_heat_swh_use.py

```diff
@@ -0,0 +1,211 @@
+from pint import Quantity
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_spaces_served_by_swh_use import (
+    get_spaces_served_by_swh_use,
+)
+from rct229.schema.config import ureg
+from rct229.schema.schema_enums import SchemaEnums
+from rct229.utils.assertions import assert_, getattr_
+from rct229.utils.jsonpath_utils import find_all
+from rct229.utils.pint_utils import ZERO
+from rct229.utils.utility_functions import (
+    find_exactly_one_schedule,
+    find_exactly_one_service_water_heating_distribution_system,
+    find_exactly_one_space,
+)
+
+SERVICE_WATER_HEATING_USE_UNIT = SchemaEnums.schema_enums[
+    "ServiceWaterHeatingUseUnitOptions"
+]
+
+REQUIRED_USE_UNIT = [
+    SERVICE_WATER_HEATING_USE_UNIT.POWER,
+    SERVICE_WATER_HEATING_USE_UNIT.VOLUME,
+]
+
+VOLUME_BASED_USE_UNIT = [
+    SERVICE_WATER_HEATING_USE_UNIT.VOLUME_PER_PERSON,
+    SERVICE_WATER_HEATING_USE_UNIT.VOLUME_PER_AREA,
+    SERVICE_WATER_HEATING_USE_UNIT.VOLUME,
+]
+
+WATER_DENSITY = 8.3452 * ureg("lb/gallon")
+WATER_SPECIFIC_HEAT = 1.001 * ureg("Btu/lb/delta_degF")
+
+
+def get_energy_required_to_heat_swh_use(
+    swh_use: dict, rmd: dict, building_segment: dict
+) -> dict[str, Quantity | None]:
+    """
+    This function calculates the total energy required to heat the SWH use over the course of a year.  Note - this function does not work for service water heating uses with use_units == "OTHER".  In this case, it will return 0.
+
+    Parameters
+    ----------
+    swh_use: dict, service_water_heating_uses
+    rmd: dict, RMD at RuleSetModelDescription level
+    building_segment: dict, building_segment
+
+    Returns
+    ----------
+    energy_required_by_space: A dict where the keys are space_ids and values are the total energy required to heat the swh_use for that space.  If a swh_use is not assigned to any spaces, the key will be "NO_SPACES_ASSIGNED; if the swh_use.use_units == 'OTHER', the total energy required will be set to None"
+
+    """
+    hourly_schedule_id = swh_use.get("use_multiplier_schedule")
+    hourly_schedule = (
+        find_exactly_one_schedule(rmd, hourly_schedule_id)
+        if hourly_schedule_id is not None
+        else None
+    )
+
+    is_heat_recovered_by_drain = swh_use.get("is_heat_recovered_by_drain", False)
+    use_units = getattr_(swh_use, "service_water_heating_uses", "use_units")
+
+    inlet_temperature_hourly_values = None  # need to be careful. When the `use_multiplier_schedule` key doesn't exist and `use_units` is not in VOLUME_BASED_USE_UNIT, this logic causes an error in line 119
```

Comment:
My understanding is that we need to set a global variable here, so later we can assign values to it if the unit is volume-based. Is this correct? @yunjoonjung-PNNL

Reply:
No, What I meant was
Let's suppose that ```use_units``` is not in the ```VOLUME_BASED_USE_UNIT```. In this scenario, the ```inlet_temperature_hourly_values``` value is ```None``` and if the ```hourly_schedule``` value is ```None```, this code will cause an error in line 124 because None can't have length (```len(inlet_temperature_hourly_values)```). 

Does this make sense to you?

Reply:
That's a good point.     ```
num_hours = (
        LeapYear.LEAP_YEAR_HOURS if is_leap_year else LeapYear.REGULAR_YEAR_HOURS
    )
```
`    hourly_values = (
        getattr_(hourly_schedule, "hourly_schedule", "hourly_values")
        if hourly_schedule is not None
        else [1] * num_hours
    )`

Reply:
Looks good to me. Thank you!

Reply:
Sure, that sounds good

Reply:
I removed the comment and added one more input in the function (RDS also updated).